### ISSUE TITLE
Sonarr Install Disabled Cron!

### ISSUE THREAD

**Issue by tjharman** [Sonarr/Sonarr#4658]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I had cron running on my Linux system.
Cron is used for all sorts of automated tasks.

After I installed Sonarr v3 to my Debian 10 system, using the provided .deb file, I noticed that none of my crontabs were running anymore.  I did some digging and I realised that on install, Sonarr **fully disabled** cron in systemd!

Logs:

```
Preparing to unpack .../archives/sonarr_3.0.6_all.deb ...
NzbDrone systemd startup detected at cron.service, stopping and disabling...
Created symlink /etc/systemd/system/cron.service → /dev/null.
Unpacking sonarr (3.0.6) ...
Setting up sonarr (3.0.6) ...
Found NzbDrone database in /home/tim/.config/NzbDrone, copying to /var/lib/sonarr.
Created symlink /etc/systemd/system/multi-user.target.wants/sonarr.service → /lib/systemd/system/sonarr.service.
```

### Expected Behavior

Installing Sonarr wouldn't set the systemd cron service to "masked"

### Steps To Reproduce

1. Have a Debian 10 system with cron installed and running (a default Debian 10 system)
2. Install sonarr using "dpkg -i sonarr_3.0.6_all.deb"
3. Verify if cron is set to masked using "systemctl status cron" - If it is masked, the bug is triggered

### Environment

```markdown
- OS: Debian 10
- Sonarr: 3.0.6
- Docker Install: No
- Using Reverse Proxy: N/A
- Browser: N/A
```

### What branch are you running?

Main

### Anything else?

I must be doing something silly, because Sonarr is obviously check for cron and disabling it on purpose as clearly shown by the lines:

```
NzbDrone systemd startup detected at cron.service, stopping and disabling...
Created symlink /etc/systemd/system/cron.service → /dev/null.
```

I just can't fathom why Sonarr would need to disable a critical linux process?

**Comment by Taloth** [comment]
Hey, that's an interesting problem.

What sonarr does during v3 install is that it tries to identify the systemd unit that nzbdrone (v2) is running as. It does so by checking the systemd unit that  nzbdrone is running in using `ps ax -o'user:20,pid,ppid,unit,args'` (You can see the relevant code [here](https://github.com/Sonarr/Sonarr/blob/6994ca720a9f5fc3a19236d1f100c705fda0e340/distribution/debian/preinst#L12))
This is because in v2 we didn't include a systemd unit in the package, so instead users individually created it as specified by various guides. Using the above code, the installer can easily determine the unit filename that the user used.

Now, my suspicion is that for v2 you weren't using systemd/upstart/initd for starting nzbdrone, but instead ran nzbdrone using a cronjob? That's definitely an odd approach, which explains why nobody ran into this edgecase before. (Also, long running cron jobs aren't advisable in general)
Naturally, cron runs it's jobs as part of it's process hierarchy, which meant that the systemd unit associated with the nzbdrone process was... cron.service.

I can probably throw in a  protection to make sure sonarr doesn't stop units outside of a pre-approved list, but I still things is an exceedingly rare scenario.

As for your particular case. The installer doesn't remove the unit, it simply disables it. So run `systemctl unmask cron.service` and `systemctl start cron.service` and it should all be back to normal.
Note that you must also remove the nzbdrone cronjob that you added in the past.

I'll keep the issue open so we can add the beforementioned protection in the install script.

**Comment by tjharman** [comment]
Thanks for the reply.
You're right, I was just running it via my own cronjob using @reboot.
I had v2 running just extracted locally in my home dir and executing it vi cron, when I found out about v3 I thought "I'll just install the .deb" because that's easier and yes, what you've described is exactly what happened.  Once I had it working fine I cleaned up my crontab (not realising cron wasn't running) and deleted the old v2 from my home dir.  It's only tonight when we had a power cut and I couldn't figure out why some things weren't running on bootup, that I realised something was amiss.

Anyway, thanks for the clear explanation - I agree it would make sense to have some sort of check in there in case it happens to someone else, but I also understand I am in the minority with the strange way I was running v2.

**Comment by markus101** [comment]
This wasn't a widespread issue and we're not planning on releasing a debian package with v4 which is now released.
